### PR TITLE
Move to partiql-lang-rust 0.3.*

### PR DESCRIPTION
**Pull request by am357**
*Description of changes:*

In addition, applies Clippy recommendations.

API Tests:

```bash
➜  ~ curl -H 'Content-Type: application/json; charset=UTF-8' \
  -H "Accept: application/json" \
  --data '{"query": "SELECT * FROM {'a': 1}"}' \
  -X POST http://localhost:8000/parse
"{\"text\":\"SELECT * FROM {a: 1}\",\"offsets\":{\"line_starts\":[0]},\"ast\":{\"Query\":{\"id\":9,\"node\":{\"with\":null,\"set\":{\"id\":8,\"node\":{\"Select\":{\"id\":7,\"node\":{\"project\":{\"id\":1,\"node\":{\"kind\":\"ProjectStar\",\"setq\":\"All\"}},\"from\":{\"id\":6,\"node\":{\"source\":{\"FromLet\":{\"id\":5,\"node\":{\"expr\":{\"Struct\":{\"id\":4,\"node\":{\"fields\":[{\"first\":{\"VarRef\":{\"id\":2,\"node\":{\"name\":{\"value\":\"a\",\"case\":\"CaseInsensitive\"},\"qualifier\":\"Unqualified\"}}},\"second\":{\"Lit\":{\"id\":3,\"node\":{\"Int64Lit\":1}}}}]}}},\"kind\":\"Scan\",\"as_alias\":null,\"at_alias\":null,\"by_alias\":null}}}}},\"from_let\":null,\"where_clause\":null,\"group_by\":null,\"having\":null}}}},\"order_by\":null,\"limit_offset\":null}}},\"locations\":{\"1\":{\"start\":0,\"end\":8},\"2\":{\"start\":15,\"end\":16},\"3\":{\"start\":18,\"end\":19},\"4\":{\"start\":14,\"end\":20},\"5\":{\"start\":14,\"end\":20},\"6\":{\"start\":9,\"end\":20},\"7\":{\"start\":0,\"end\":20},\"8\":{\"start\":0,\"end\":20},\"9\":{\"start\":0,\"end\":20}}}"% ➜  ~ curl -H 'Content-Type: application/json; charset=UTF-8' \
  -H "Accept: application/json" \
  --data '{"query": "SELECT * FROM {'a': 1}"}' \
  -X POST http://localhost:8000/explain
"\"LogicalPlan { nodes: [ProjectAll, Scan(Scan { expr: TupleExpr(TupleExpr { attrs: [VarRef(CaseInsensitive(\\\"a\\\"))], values: [Lit(1)] }), as_key: \\\"_1\\\", at_key: None }), Sink], edges: [(OpId(2), OpId(1), 0), (OpId(1), OpId(3), 0)] }\""% ➜  ~ curl -H 'Content-Type: application/json; charset=UTF-8' \
  -H "Accept: application/json" \
  --data '{"query": "SELECT * FROM {'a': 1}"}' \
  -X POST http://localhost:8000/explain
"\"LogicalPlan { nodes: [ProjectAll, Scan(Scan { expr: TupleExpr(TupleExpr { attrs: [VarRef(CaseInsensitive(\\\"a\\\"))], values: [Lit(1)] }), as_key: \\\"_1\\\", at_key: None }), Sink], edges: [(OpId(2), OpId(1), 0), (OpId(1), OpId(3), 0)] }\"
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jpschorr — APPROVED** [review]

File: .github/workflows/ci_build_test.yml [hunk 1]
```diff
@@ -34,10 +34,7 @@ jobs:
           command: build
           args: --verbose --workspace
       - name: Cargo Test
-        uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --verbose --workspace
+        run: cargo test --verbose --workspace -- -Z unstable-options
```
Comment (alancai98) [inline .github/workflows/ci_build_test.yml:37]: Think the error you're getting in the CI seems similar to https://github.com/partiql/partiql-lang-rust/pull/322/files.

Might need to use the nightly toolchain?

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -26,10 +26,7 @@ jobs:
           toolchain: nightly-2022-08-11
```
Comment (alancai98) [inline .github/workflows/coverage.yml]: And for the coverage CI, probably need to upgrade the nightly version to a more recent version such as 2023-03-09

File: .github/workflows/ci_build_test.yml (hunk 1 above)
Comment (am357) [reply .github/workflows/ci_build_test.yml:37]: +1, submitted a new commit, let's see.

Comment (alancai98) [reply .github/workflows/ci_build_test.yml:37]: Actually, nightly might not be needed for the build/test. Think you'll need it for the coverage step though.

I'd update the rust toolchain to `dtolnay/rust-toolchain` (like in [partiql-lang-rust](https://github.com/partiql/partiql-lang-rust/blob/main/.github/workflows/ci_build_test.yml#L32-L36)) which is still maintained.

Comment (am357) [reply .github/workflows/ci_build_test.yml:37]: Looks like with the new changes we're getting the successful builds; but I think it's better to make it consistent w/the other project. Let me do that to see the outcome.

**Pull request merged**